operators/test/e2e/kb/keystore_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package kb
+
+import (
+	"testing"
+
+	kbtype "github.com/elastic/cloud-on-k8s/operators/pkg/apis/kibana/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/operators/test/e2e/framework"
+	"github.com/elastic/cloud-on-k8s/operators/test/e2e/framework/elasticsearch"
+	"github.com/elastic/cloud-on-k8s/operators/test/e2e/framework/kibana"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestUpdateKibanaSecureSettings(t *testing.T) {
+	k := framework.NewK8sClientOrFatal()
+
+	// user-provided secure settings secret
+	secureSettingsSecretName := "secure-settings-secret"
+	secureSettings := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secureSettingsSecretName,
+			Namespace: framework.Namespace,
+		},
+		Data: map[string][]byte{
+			// this needs to be a valid configuration item, otherwise Kibana refuses to start
+			"logging.verbose": []byte("true"),
+		},
+	}
+
+	// set up a 1-node Kibana deployment with secure settings
+	name := "test-kb-keystore"
+	esBuilder := elasticsearch.NewBuilder(name).
+		WithESMasterDataNodes(1, elasticsearch.DefaultResources)
+	kbBuilder := kibana.NewBuilder(name).
+		WithNodeCount(1).
+		WithKibanaSecureSettings(secureSettings.Name)
+
+	framework.TestStepList{}.
+		// create secure settings secret
+		WithStep(framework.TestStep{
+			Name: "Create secure settings secret",
+			Test: func(t *testing.T) {
+				// remove if already exists (ignoring errors)
+				_ = k.Client.Delete(&secureSettings)
+				// and create a fresh one
+				err := k.Client.Create(&secureSettings)
+				require.NoError(t, err)
+			},
+		}).
+		// create the cluster
+		WithSteps(esBuilder.InitTestSteps(k)).
+		WithSteps(kbBuilder.InitTestSteps(k)).
+		WithSteps(esBuilder.CreationTestSteps(k)).
+		WithSteps(kbBuilder.CreationTestSteps(k)).
+		WithSteps(framework.CheckTestSteps(esBuilder, k)).
+		WithSteps(framework.CheckTestSteps(kbBuilder, k)).
```

Comment:
IIUC you can't use `framework.Run` here because you want to inject a step in front of the standard set of steps. 

I am wondering if it is worth separating step sequence construction and execution more. 

```go

steps := framkwork.TestStepList{
                    framework.TestStep{
			Name: "Create secure settings secret",
			Test: func(t *testing.T) {
				// remove if already exists (ignoring errors)
				_ = k.Client.Delete(&secureSettings)
				// and create a fresh one
				err := k.Client.Create(&secureSettings)
				require.NoError(t, err)
			}
}
steps = append(steps, framework.RunSequence(t, testStepFn, esBuilder, kbBuilder)...)
steps.RunSequential(t)
```

Reply:
I think it's worth it.

Reply:
I tried to address this in 24a02a0.
I'm not sure how it's important to share the same instance of the K8SClient in all steps.